examples/custom-service.mdx

```diff
@@ -20,71 +38,163 @@ edition = "2021"
-to create a wrapper around our `PgPool`.
+### Getting Started
+To get started, we need to return a wrapper struct from our `shuttle_service::main` function that 
+implements `shuttle_service::Service`, which we can declare like so:
 
-```rust
-struct PoolService {
-    pool: sqlx::PgPool,
+```rust lib.rs
+pub struct CustomService {
+    discord_bot:
+        FrameworkBuilder<Data, Box<(dyn std::error::Error +
+         std::marker::Send + Sync + 'static)>>,
+    router: SyncWrapper<Router>,
 }
 ```
 
 Then we need to implement `shuttle_service::Service` for our wrapper. If you need to bind to an address,
 for example if you're implementing service for an HTTP server, you can use the `addr` argument from `bind`.
 You can only have one HTTP service bound to the `addr`, but you can start other services that don't rely on 
-binding to a socket. A common usecase is starting an HTTP server like `axum`, alongside a `serenity` bot 
-that doesn't need to bind to a socket. Then you would include both in the same wrapper struct, and start 
-them as normal in the `bind` function.
+binding to a socket.
+
+We can do do this by simply adding an implementation for our wrapper struct containing a function called `start`,
+which will contain all of our tasks that we want to run and simply run them all together in one `tokio::select!` macro
+(currently, this code doesn't work due to no base implementation - we will implement this later on):
 
 ```rust lib.rs
 #[shuttle_service::async_trait]
-impl shuttle_service::Service for PoolService {
+impl shuttle_service::Service for CustomService {
     async fn bind(
         mut self: Box<Self>,
-        _addr: std::net::SocketAddr,
+        addr: std::net::SocketAddr,
     ) -> Result<(), shuttle_service::error::Error> {
-        self.start().await?;
+        self.start(addr).await.expect("Something went wrong!");
 
         Ok(())
     }
 }
 ```
+### Commands/Routing
+Before we can actually run the program, we will need to set up the commands and routing that it needs
+ before we can add them to the struct implementation. Let's do that now:
+
+```rust commands.rs
+use poise::serenity_prelude as serenity;
+
+// this is a blank struct initialised in lib.rs and then imported here
+use crate::Data;
+
+type Error = Box<dyn std::error::Error + Send + Sync>;
+type Context<'a> = poise::Context<'a, Data, Error>;
+
+#[poise::command(slash_command, prefix_command)]
+pub async fn age(
+    ctx: Context<'_>,
+    #[description = "Selected user"] user: Option<serenity::User>,
+) -> Result<(), Error> {
+    let u = user.as_ref().unwrap_or_else(|| ctx.author());
+    let response = format!("{}'s account was created at {}", u.name, u.created_at());
+    ctx.say(response).await?;
+    Ok(())
+}
+```
+
+```rust router.rs
+use axum::http::StatusCode;
+use axum::response::IntoResponse;
+use axum::{routing::get, Router};
+
+pub fn router() -> Router {
```

Comment:
Nit-pick: maybe this should be named `build_router` or something more descriptive, or just `app`?

Reply:
Amended